.github/workflows/changelog.py

```diff
@@ -0,0 +1,202 @@
+
+
+# If "(chore)" or "(docs)" is appended to the PR title, it indicates that we don't want to log this change.
+if pr_title.endswith("(chore)") or pr_title.endswith("(docs)"):
+    print("Skipping logging this change as it's a chore or docs update")
+    sys.exit(0)
+
+
+def find_module_info(module_name):
+    """
+    Helper function to load module meta info. With current setup, can't really just import
+    the module and call `mod.info`, as the module does the heavy work on initialization.
+    But that's good - we avoid installing and importing MultiQC here, and the action runs faster.
+    """
+    module_name = module_name.lower()
+    modules_dir = Path("../../multiqc/modules")
+    py_path = None
+    for dir_name in os.listdir(modules_dir):
+        if dir_name.lower() == module_name:
+            module_dir = modules_dir / dir_name
+            py_path = module_dir / f"{dir_name}.py"
+            if not py_path.exists():
+                print(f"Folder for {module_name} exists, but doesn't have a {py_path} file", file=sys.stderr)
+                sys.exit(1)
+            break
+
+    if not py_path:  # Module not found
+        return None
+    with py_path.open("r") as f:
+        contents = f.read()
+    if not (m := re.search(r'name="([^"]+)"', contents)):
+        return None
+    name = m.group(1)
+    if not (m := re.search(r'href="([^"]+)"', contents)):
+        return None
+    url = m.group(1)
+    if not (m := re.search(r'info="([^"]+)"', contents)):
+        if not (m := re.search(r'info="""([^"]+)"""', contents)):
+            return None
+    info = m.group(1)
+    # Reduce consecutive spaces and newlines.
+    info = re.sub(r"\s+", " ", info)
+    return {"name": name, "url": url, "info": info}
+
+
+# Determine the type of the PR: new module, module update, or core update.
+mod = None
+section = "### MultiQC updates"  # Default section for non-module (core) updates.
+if pr_title.lower().startswith("new module: "):
+    # PR introduces a new module.
+    section = "### New modules"
+    module_name = pr_title.split(":")[1].strip()
+    mod = find_module_info(module_name)
+    if not mod:
+        # That should normally never happen because the other CI would fail and block
+        # merging of the PR.
+        print(
+            f"Cannot load a module with name {module_name}",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+else:
+    # Checking if it's an existing module update.
+    maybe_mod_name = pr_title.split(":")[0]
+    mod = find_module_info(maybe_mod_name)
+    if mod is not None:
+        section = "### Module updates"
+        pr_title = pr_title.split(":")[1].strip().capitalize()
+
+# Now that we determined the PR type, preparing the change log entry.
+pr_link = f"([#{pr_number}]({REPO_URL}/pull/{pr_number}))"
+if section == "### New modules":
+    new_lines = [
+        f"- [**{mod['name']}**]({mod['url']})\n",
+        f"  - {mod['name']} {mod['info']}\n",
```

Comment:
I have a feeling that GitHub markdown rendering needs 4 spaces for list indentation. Maybe check, but I think that's the case.

Reply:
Our changelog file uses 2 spaces there: https://raw.githubusercontent.com/ewels/MultiQC/master/CHANGELOG.md

As long as they are not tabs!